### DISCUSSION TITLE
Allow to use language prefix in URLs instead of query parameter

### DISCUSSION THREAD

**Discussion by der-On** [bagisto/bagisto#6624] [General]
For SEO purposes it would be good to use a language prefix in URLs for mulilingual shops.

Required changes:

- add a setting for enabling language prefixes in the backend or as a config option.
- if language prefixes are enabled wrap all shop frontend routes in a prefixed group
- extend the locale middleware to either use the language prefix (if enabled) or the query parameter
- extend the route() helper to automaticly prepend the language prefix of the current locale if language prefixes are enabled
- ... possibly other things I'm not aware of

**Comment by Sanjaybhattwebkul** [comment]
Hi @der-On 
**Thanks** for your suggestion. we will **implement** it and will reflect it in the **upcoming release**.

**Comment by devansh-webkul** [comment]
- This can be done by adding multiple channels.
- Mapping the subdomain to channel.
- Then assigning the default locale to specific channel.

**Reply by der-On** [reply]
Not sure if this proposed solution is ideal as it creates a lot of complexity and I think channels have another purpose.
